src/domain/event/utils.ts

```diff
@@ -125,3 +130,17 @@ export const orderOccurrencesByDate = (
   }
   return 0;
 };
+
+export const getRealKeywords = (
+  eventData: EventQuery
+): Keyword[] | undefined => {
+  const { additionalCriteria, categories } = eventData.event || {};
+  return eventData?.event?.keywords.filter((keyword) => {
+    return !(
+      categories?.find((category) => category.id === keyword.id) ||
+      additionalCriteria?.find(
+        (additionalCriteria) => additionalCriteria.id === keyword.id
+      )
+    );
+  });
+};
```

Comment:
This could at least include a description. It's hard to understand what the "real keywords" are and what is their purpose - Where are the used or needed?

Wouldn't it be faster and better looking if the filter would be called twice, like:

```
  return eventData?.event?.keywords
    ?.filter((keyword) => !(categories?.find((category) => category.id === keyword.id)))
    ?.filter((keyword) => !(categories?.find((additionalCriteria) => additionalCriteria.id === keyword.id)))
```

Also, doesn't `keywords.filter` need a null check also?

Reply:
Yes I agree

Reply:
I think the best practise with filter functions is that it should always do only 1 thing... Like with every function :) I've read several articles about this but now I couldn't find any other but this, which ain't very good: https://advancedweb.hu/best-practices-on-how-to-work-with-collections-in-javascript/

Reply:
Yeah those should definitely be move to separate filters at least

Reply:
Improved the function by combining those arrays first
```
// Get keywords that are not already included in additionalCriteria or categories.
export const getRealKeywords = (
  eventData: EventQuery
): Keyword[] | undefined => {
  const { additionalCriteria, categories } = eventData.event || {};

  // Combine other keywords into a single array to make filtering simpler
  const otherKeywords = [...(additionalCriteria ?? []), ...(categories ?? [])];

  return eventData?.event?.keywords.filter(
    (keyword) => !otherKeywords?.find((category) => category.id === keyword.id)
  );
};
```